### ISSUE TITLE
Port `twisted.internet.abstract` to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6163]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6163|
|Type|enhancement|
|Created|2012-10-26 14:33:56Z|
|Branch|https://github.com/twisted/twisted/tree/abstract-py3-6163|

`twisted.internet.abstract` should be fully ported to Python 3, along with `twisted.test.test_abstract`.

<details><summary>Searchable metadata</summary>

```
trac-id__6163 6163
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_abstract_py3_6163 branches/abstract-py3-6163
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1351262036000000 1351262036000000
changetime__1352145257000000 1352145257000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36272]) Branching to 'abstract-py3-6163'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/abstract-py3-6163

Just added `twisted.test.test_abstract`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [36348]) Merge abstract-py3-6163.

Author: itamar
Review: exarkun
Fixes: [#6163](https://github.com/twisted/twisted/issues/6163)

twisted.internet.abstract now officially ported to Python 3.
